sklearn/utils/stats.py

```diff
@@ -7,30 +7,35 @@
 
 
 def _weighted_percentile(array, sample_weight, percentile=50):
-    """Compute weighted percentile
+    """Compute the lower value(s) at a given weighted percentile.
 
-    Computes lower weighted percentile. If `array` is a 2D array, the
-    `percentile` is computed along the axis 0.
+    If `array` is a 2D array, the `values` are selected along axis 0.
 
         .. versionchanged:: 0.24
             Accepts 2D `array`.
 
+        .. versionchanged:: 1.6
+            Supports handling of `NaN` values. For `NaN` inputs, their corresponding
+            weights are set to 0 in `sample_weight`.
+            Percentiles that point to `NaN` values are redirected to the next lower
+            value if it exists.
+
     Parameters
     ----------
     array : 1D or 2D array
         Values to take the weighted percentile of.
 
     sample_weight: 1D or 2D array
-        Weights for each value in `array`. Must be same shape as `array` or
-        of shape `(array.shape[0],)`.
+        Weights for each value in `array`. Must be same shape as `array` or of shape
+        `(array.shape[0],)`.
 
     percentile: int or float, default=50
-        Percentile to compute. Must be value between 0 and 100.
+        Percentile to compute. Must be between 0 and 100.
 
     Returns
     -------
-    percentile : int if `array` 1D, ndarray if `array` 2D
-        Weighted percentile.
+    values : int if `array` 1D, ndarray if `array` 2D
+        Lower value(s) at a given weighted percentile.
```

Comment:
```suggestion
    percentile : int if `array` 1D, ndarray if `array` 2D
        Value of weighted percentile at the requested level.
```

The previous name was actually more correct (e.g. as per the definitions of percentile (scores) and percentile rank in https://en.wikipedia.org/wiki/Percentile). I suppose the need to rename this output variable stems from the conflict with the badly named `percentile` input argument: the latter should better be named `percentile_level` or `percentile_rank`. Since it's a private function, I would not be opposed to renaming the input argument instead.

Reply:
Thanks for hinting me to the Wikipedia article. However, it doesn't help me form an opinion about how to best name the variables in our case. I grasp that the data is subdivided into 100 percentiles, though I'm confused by the distinction that the unit is not a percentage but the unit of the data itself. My current understanding is that a percentile is an integer between 1 and 100, not a specific value (or several if 2D) from the input array (or "score" as they refer to it, for reasons I'm not quite sure of). I’ve also seen charts like [this one](https://www.cdc.gov/growthcharts/data/set1clinical/cj41l017.pdf), where you can find which percentile a specific value falls on for comparison reasons. 

From this, I understand this algorithm like this: we give it a percentile/percentage of interest and it selects the values from the data that are closest to this percentile from below.

If working through this PR with me is becoming too time-consuming due to my lack of statistical knowledge, I understand if you don't answer to all my questions or if you'd prefer to take it over.

Reply:
fyi: wikipedia is not the best reference for quantiles